src/main/java/at/hannibal2/skyhanni/features/rift/area/mountaintop/SunGeckoHelper.kt

```diff
@@ -0,0 +1,266 @@
+        }
+
+        val health = "$healthColor$displayHealthLeft§f/§a$displayTotalHealth§c❤"
+        display.add("§eSun Gecko $health")
+        display.add("$actionBarFormatted §e§lCombo: x$combo")
+
+        //this is just a total guess but it looks right enough
+        //i think its inconsistent because of how often the action bar updates
+        var expiryTime = 5.seconds + 200.milliseconds
+        if (modifiers.contains(MODIFIERS.COLLECTIVE)) {
+            expiryTime += (modifiers.count() * 200).milliseconds
+        }
+        val timeLeft = timeSinceLastHit + expiryTime
+        if (timeLeft.timeUntil().inWholeMilliseconds > expiryTime.inWholeMilliseconds - 800.milliseconds.inWholeMilliseconds ||
+            timeLeft.isInPast()) {
+            display.add("§aCombo Timer: ${expiryTime.format()}/${expiryTime.format()}")
+        } else {
+            display.add("§aCombo Timer: ${timeLeft.timeUntil().format(showMilliSeconds = true)}/${expiryTime.format()}")
+        }
+
+        if (config.showModifiers) {
+            display.add("§6Modifiers:")
+            for (modifier in modifiers) {
+                display.add("§e${modifier.name.allLettersFirstUppercase()}")
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onMobSpawn(event: MobEvent.Spawn) {
+        if (!event.mob.name.contains("Sun Gecko")) return
+        if (event.mob.name.contains("?") && config.highlightFakeBoss) {
+            RenderLivingEntityHelper.setEntityColorWithNoHurtTime(
+                event.mob.baseEntity,
+                Color.RED.addAlpha(80),
+            ) { config.highlightFakeBoss }
+        } else if (config.highlightRealBoss) {
+            RenderLivingEntityHelper.setEntityColorWithNoHurtTime(
+                event.mob.baseEntity,
+                Color.GREEN.addAlpha(80),
+            ) { config.highlightRealBoss }
+        }
+
+    }
+
+
+    @SubscribeEvent
+    fun onGuiRender(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (!inTimeChamber) return
+
+        var nearestEntityDistance = 100.0
+        var nearestEntity: Entity? = null
+        for (entity in EntityUtils.getAllEntities()) {
+            sunGeckoName.matchMatcher(entity.name) {
+                if (entity.distanceToPlayer() < nearestEntityDistance) {
+                    nearestEntityDistance = entity.distanceToPlayer()
+                    nearestEntity = entity
+                }
+            }
+        }
+
+        if (nearestEntity == null) return
+
+        sunGeckoName.matchMatcher(nearestEntity?.name.orEmpty()) {
+            val health = group("healthLeft")?.toIntOrNull() ?: -1
+            if (health > healthLeft) {
+                onFirstPhase = false
+                modifiers.add(MODIFIERS.REVIVAL)
+            }
+            healthLeft = health
+            totalHealth = group("totalHealth")?.toIntOrNull() ?: 250
+            healthColor = group("healthColor")?.toString() ?: "§a"
+        }
```

Comment:
This whole logic should be replaced with mob detection stuff. Eg: Grab the enitity once (idk how to always find the correct one in the chamber and not one from someone else), then update the health in a tick event (since we want to have as few code as possible in render events, since those scale the worst because of the frequent calls).

Reply:
think ive done this